library/wizard/test/wizard_test.rb

```diff
@@ -0,0 +1,62 @@
+#! /usr/bin/env rspec
+
+require_relative "test_helper"
+
+Yast.import "Wizard"
+
+describe Yast::Wizard do
+  subject { described_class }
+
+  describe ".CreateDialog" do
+    before do
+      allow(Yast::UI).to receive(:OpenDialog)
+    end
+
+    it "sets product name in UI" do
+      expect(Yast::UI).to receive(:SetProductName)
+
+      subject.CreateDialog
+    end
+
+    it "opens Dialog" do
+      expect(Yast::UI).to receive(:OpenDialog)
+
+      subject.CreateDialog
+    end
+  end
+
+  describe ".SetHelpText" do
+    it "calls UI Wizard command SetHelpText" do
+      expect(Yast::UI).to receive(:WizardCommand).with(term(:SetHelpText, "test")).and_return(true)
+
+      subject.SetHelpText("test")
+    end
+
+    it "calls ChangeWidget if Wizard Command failed" do
+      allow(Yast::UI).to receive(:WizardCommand).and_return(false)
+      expect(Yast::UI).to receive(:ChangeWidget).with(anything, :HelpText, "test")
+
+      subject.SetHelpText("test")
+    end
+  end
+
+  describe ".UserInput" do
+    it "returns result of UI UserInput" do
+      allow(Yast::UI).to receive(:UserInput).and_return(:ok)
```

Comment:
NP: I admit that I prefer to use the allow outside of `it`, usually in a `before` block and then use `context` to set the return value. Something like

```ruby
describe ".UserInput" do
  before do
    allow(Yast::UI).to receive(:UserInput).and_return(user_input)
  end

  context "when user input is :ok" do
    let(:user_input) { :ok }

    it "returns result of UI UserInput" do
      expect(subject.UserInput).to eq :ok
    end
  do

  context "when user accepts" do
    let(:user_input) { :accept }

    it "returns :next" do
      expect(subject.UserInput).to eq :next
    end
  end

  context "when user cancels" do
    let(:user_input) { :cancel }

    it "returns :back" do
      expect(subject.UserInput).to eq :back
    end
  end
end
```

Reply:
I know that there is a bit disagreement, as I prefer more explicit tests which is from first sight visible what each api call returns and this style make it a less obvious for me. My code is just two related calls, one sets user input and second just check return value without need to check another contexts, lets and other stuff.